sonarqube/resource_sonarqube_new_code_periods_test.go

```diff
@@ -0,0 +1,321 @@
+	return fmt.Sprintf(`
+        resource "sonarqube_new_code_periods" "%[1]s" {
+			type = "NUMBER_OF_DAYS"
+			value = "5"
+        }`, rnd)
+}
+
+func TestAccSonarqubeNewCodePeriodsGlobalNumberOfDays(t *testing.T) {
+	rnd := generateRandomResourceName()
+	name := "sonarqube_new_code_periods." + rnd
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSonarqubeNewCodePeriodsGlobalNumberOfDays(rnd),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "type", "NUMBER_OF_DAYS"),
+					resource.TestCheckResourceAttr(name, "value", "5"),
+				),
+			},
+		},
+	})
+}
+
+func testAccSonarqubeNewCodePeriodsBranchPreviousVersion(rnd string) string {
+	return fmt.Sprintf(`
+	    resource "sonarqube_project" "%[1]s" {
+			name = "%[1]s"
+			project = "%[1]s"
+			visibility = "public"
+		}
+
+        resource "sonarqube_new_code_periods" "%[1]s" {
+			branch = "main"
+			project = sonarqube_project.%[1]s.project
+			type = "PREVIOUS_VERSION"
+        }`, rnd)
+}
+
+func TestAccSonarqubeNewCodePeriodsBranchPreviousVersion(t *testing.T) {
+	rnd := generateRandomResourceName()
+	name := "sonarqube_new_code_periods." + rnd
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSonarqubeNewCodePeriodsBranchPreviousVersion(rnd),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "type", "PREVIOUS_VERSION"),
+					resource.TestCheckResourceAttr(name, "branch", "main"),
+					resource.TestCheckResourceAttr(name, "project", rnd),
+				),
+			},
+		},
+	})
+}
+
+func testAccSonarqubeNewCodePeriodsBranchNumberOfDays(rnd string) string {
+	return fmt.Sprintf(`
+	    resource "sonarqube_project" "%[1]s" {
+			name = "%[1]s"
+			project = "%[1]s"
+			visibility = "public"
+		}
+
+        resource "sonarqube_new_code_periods" "%[1]s" {
+			branch = "main"
+			project = sonarqube_project.%[1]s.project
+			type = "NUMBER_OF_DAYS"
+			value = "5"
+        }`, rnd)
+}
+
+func TestAccSonarqubeNewCodePeriodsBranchNumberOfDays(t *testing.T) {
+	rnd := generateRandomResourceName()
+	name := "sonarqube_new_code_periods." + rnd
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSonarqubeNewCodePeriodsBranchNumberOfDays(rnd),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "type", "NUMBER_OF_DAYS"),
+					resource.TestCheckResourceAttr(name, "branch", "main"),
+					resource.TestCheckResourceAttr(name, "project", rnd),
+					resource.TestCheckResourceAttr(name, "value", "5"),
+				),
+			},
+		},
+	})
+}
+
+// Note sure how to get a specific analysis UUID on the fly in a test environment
```

Comment:
yah me either. Are there even apis we could call to get one? Clicking through the web_api docs i dont see one.

Reply:
I did find [this API](https://next.sonarqube.com/sonarqube/web_api/api/project_analyses), but then we'd also have to run an analysis too which would mean setting up dummy code, configuring the project fully, and calling some other API to run an analysis...